src/TorchSharp/JIT/ScriptModule.cs

```diff
@@ -0,0 +1,309 @@
+                            continue;
+
+                        case Tensor tensor: {
+                                var t = tensor.to(dtype);
+                                field.SetValue(this, t);
+                                ConditionallyRegisterBuffer(fieldName, t);
+                                break;
+                            }
+                        }
+                    }
+
+                    return this;
+                }
+
+#if false   // These functions "work," but the native code doesn't seem to find any interesting information.
+
+                [DllImport("LibTorchSharp")]
+                private static extern IntPtr THSJIT_Module_getInputType(HType module, int index);
+
+                public Type GetInputType(int index)
+                {
+                    var type = new Type(THSJIT_Module_getInputType(handle, index), Type.TypeKind.AnyType);
+
+                    return GetType(type);
+                }
+
+                [DllImport("LibTorchSharp")]
+                private static extern IntPtr THSJIT_getOutputType(HType module, int index);
+
+                public Type GetOutputType(int index)
+                {
+                    var type = new Type(THSJIT_getOutputType(handle, index), Type.TypeKind.AnyType);
+
+                    return GetType(type);
+                }
+
+                private Type GetType(Type type)
+                {
+                    switch (type.Kind) {
+                    case Type.TypeKind.TensorType:
+                        var dynamic = type.AsTensorType();
+                        type.Dispose();
+                        return dynamic;
+                    //case Type.TypeKind.DimensionedTensorType:
+                    //    var tensor = type.AsTensorType();
+                    //    type.Dispose();
+                    //    return tensor;
+                    default:
+                        return type;
+                    }
+                }
+#endif
+
+                [DllImport("LibTorchSharp")]
+                private static extern IntPtr THSJIT_Module_forward(HType module, IntPtr tensors, int length);
+
+                public unsafe override Tensor forward(Tensor tensor)
+                {
+                    using (var parray = new PinnedArray<IntPtr>()) {
+                        IntPtr tensorRefs = parray.CreateArray(new[] { tensor.Handle });
+                        return new Tensor(THSJIT_Module_forward(handle, tensorRefs, parray.Array.Length));
+                    }
+                }
+            }
+
+            [DllImport("LibTorchSharp")]
+            private static extern IntPtr THSJIT_load(string filename);
+
+            public static ScriptModule load(string filename)
+            {
+                if (!System.IO.File.Exists(filename))
+                    throw new System.IO.FileNotFoundException(filename);
+                return new ScriptModule(THSJIT_load(filename));
```

Comment:
>THSJIT_load(filename) [](http://example.com/codeflow?start=40&length=21)

What happen if I pass a file pass that is exist but not containing a valid script? does the native side somehow throw? how we recognize this failing case?

Reply:
It will blow up in some fashion. We're just interfacing with the libtorch support for TorchScript, so whatever it does. I'll add a negative unit test.